chore(Ollama Chat Model Node): Update @langchain/ollama to 1.0.3 to for separation of thinking

johanatandromeda opened the pull request:
## Summary

With the newest ollama, the thinking and output are sent in different tags. In the previous version of @langchain/ollama, they were concatenated without a way to separate the thinking and AI output in N8N.

## Related Linear tickets, Github issues, and Community forum posts

https://community.n8n.io/t/ollama-thinking-included-in-response-with-1-123-5-and-ollama-0-13-2/234224/3

## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) <!--
   **Remember, the title automatically goes into the changelog.
   Use `(no-changelog)` otherwise.**
-->
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. <!--
   A bug is not considered fixed, unless a test is added to prevent it from happening again.
   A feature is not complete without tests.
-->
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)
- [x] I have seen this code, I have run this code, and I take responsibility for this code.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/n8n-io/n8n?pullRequest=23098) <br/>All committers have signed the CLA.

dgarvis commented:
thank you for finding this. N8N + Ollama had become unusable. Hope they approve it into the main branch soon.

johanatandromeda commented:
Yes, but upgrade to language chain 1.0.3 separates them again. The thinking is not visible in n8n though, even if sent from language chain in a separate field.

johanatandromeda commented:
@diego-treitos,  I patch the Docker image until the PR is accepted.

This is my Dockerfile (removed the Timeout patch from https://github.com/Piggeldi2013/n8n-timeout-patch).

```
FROM n8nio/n8n:latest
USER root
RUN mkdir -p /opt/extra && \
    npm --prefix /opt/extra install @langchain/ollama@1.0.3 && \
    cp /opt/extra/node_modules/@langchain/ollama/dist/utils.* $(dirname $(find /usr/local/lib/ -name utils.js | grep ollama)) && \
    chown -R node:node /opt/extra
USER node
```

emiroztrk commented:
Is there an estimated timeline for when this will be approved and merged?
Managing the process with temporary workarounds has become quite challenging.
Hopefully, it will be merged as soon as possible.

WilfridoCO commented:
> @johanatandromeda you're my hero

For those who want to update and are newies like me, this is the way to update your docker-n8n:

First make a new directory and create a "Dockerfile" with this:

FROM docker.n8n.io/n8nio/n8n
USER root

RUN mkdir -p /opt/extra && \
    npm --prefix /opt/extra install @langchain/ollama@1.0.3 && \
    cp /opt/extra/node_modules/@langchain/ollama/dist/utils.* $(dirname $(find /usr/local/lib/ -name utils.js | grep ollama)) && \
    chown -R node:node /opt/extra

USER node

After that, open a terminal and go to the new directory, then use:
sudo docker build -t n8n-ollama .

Finally stop, remove and rerun your docker:
sudo docker stop n8n

sudo docker rm n8n

sudo docker run -d \
  --name n8n \
  -p 5678:5678 \
  -e NODE_FUNCTION_ALLOW_EXTERNAL="*" \
  -v "IN CASE YOU HAVE A DIRECTORY AS VOLUME" \
  --restart=always \
  n8n-ollama

Thank you again @johanatandromeda!

OlegIvaniv reviewed (dismissed):
Thanks for this, @johanatandromeda!

Chappie74 commented:
+1 on this too please. 
Updated and my entire conversation is bogged with the think :(

JayAndre562 commented:
Any updates on this, this is such a headache righ tnow

johanatandromeda commented:
@garritfra I tried to update pnpm-lock.yml but managed to close the PR instead.

johanatandromeda commented:
@garritfra I think I solved it.

garritfra commented:
@johanatandromeda thanks! :)

jhedin-csam committed:
chore(Ollama Chat Model Node): Update @langchain/ollama to 1.0.3 to for separation of thinking

With the newest ollama, the thinking and output are sent in different tags. In the previous version of @langchain/ollama, they were concatenated without a way to separate the thinking and AI output in N8N.

Co-authored-by: Johan Hedin <johan.hedin@omda.com>
Made-with: Cursor

garritfra committed:
Merge branch 'master' into master

garritfra committed:
Merge branch 'master' into master

garritfra committed:
Merge branch 'master' into master

garritfra committed:
Merge branch 'master' into master

garritfra committed:
Merge branch 'master' into master

garritfra commented:
Sorry this took so long. We had some issues with e2e tests on community PRs, which are finally fixed! 🙂
